### PR TITLE
tril_indices

### LINKED ISSUE #18039: tril_indices
Working on (#15024 -> )

### PR DESCRIPTION
**Pull request by Sarvesh-Kesharwani**
Closed #18039

### CODE REVIEW

**Comment by Sarvesh-Kesharwani** [comment]
pycharm64_Jz53Ojlwa4

File: ivy_tests/test_ivy/helpers/function_testing.py [hunk 1]
```diff
@@ -805,7 +805,6 @@ def arrays_to_numpy(x):
         rtol = _get_framework_rtol(rtol, ivy.backend)
     if isinstance(atol, dict):
         atol = _get_framework_atol(atol, ivy.backend)
-
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/function_testing.py:808]: Could you please revert unrelated changes to the issue? Thanks!

**Review by CatB1t — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/paddle/tensor/creation.py [hunk 2]
```diff
@@ -114,3 +114,11 @@ def tril(x, diagonal=0, name=None):
 @to_ivy_arrays_and_back
 def triu(x, diagonal=0, name=None):
     return ivy.triu(x, k=diagonal)
+
+
+@with_supported_dtypes({"2.5.0 and below": ("int32", "int64")}, "paddle")
+@to_ivy_arrays_and_back
+def tril_indices(row, col=None, offset=0, dtype="int64"):
```
Comment (hmahmood24) [inline ivy/functional/frontends/paddle/tensor/creation.py:121]: The `col` parameter is not optional and so it shouldn't have a default value of `None` as indicated in the official docs

File: ivy/functional/frontends/paddle/tensor/creation.py [hunk 3]
```diff
@@ -114,3 +114,11 @@ def tril(x, diagonal=0, name=None):
 @to_ivy_arrays_and_back
 def triu(x, diagonal=0, name=None):
     return ivy.triu(x, k=diagonal)
+
+
+@with_supported_dtypes({"2.5.0 and below": ("int32", "int64")}, "paddle")
+@to_ivy_arrays_and_back
+def tril_indices(row, col=None, offset=0, dtype="int64"):
+    arr = ivy.tril_indices(row, col, offset)
+    arr = ivy.astype(arr, ivy.as_native_dtype(dtype))
```
Comment (hmahmood24) [inline ivy/functional/frontends/paddle/tensor/creation.py:123]: Again, we shouldn't be passing in `ivy.as_native_dtype` here, instead just `dtype`

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_creation.py [hunk 4]
```diff
@@ -420,3 +420,41 @@ def test_paddle_triu(
...
+# tril_indices
+@handle_frontend_test(
+    fn_tree="paddle.tril_indices",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("integer")
+    ),
+    dtype=helpers.get_dtypes("valid", full=False),
+    row=st.integers(min_value=1, max_value=5),
+    col=st.integers(min_value=1, max_value=5),
+    offset=st.integers(min_value=0, max_value=4),
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_creation.py:434]: `offset` should be tested for negative values as well

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @Sarvesh-Kesharwani, requested changes for you to look into. Thanks! :)

**Pull request closed without merging**